# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
#
#   Licensed under the Apache License, Version 2.0 (the "License").
#   You may not use this file except in compliance with the License.
#   A copy of the License is located at
#
#       http://www.apache.org/licenses/LICENSE-2.0
#
#   or in the "license" file accompanying this file. This file is distributed
#   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
#   express or implied. See the License for the specific language governing
#   permissions and limitations under the License.

diff -ruN byte-buddy-byte-buddy-1.12.6/byte-buddy-dep/src/main/java/net/bytebuddy/dynamic/ClassFileLocator.java byte-buddy-byte-buddy-1.12.6-patched/byte-buddy-dep/src/main/java/net/bytebuddy/dynamic/ClassFileLocator.java
--- byte-buddy-byte-buddy-1.12.6/byte-buddy-dep/src/main/java/net/bytebuddy/dynamic/ClassFileLocator.java	2021-12-20 02:30:58.000000000 -0800
+++ byte-buddy-byte-buddy-1.12.6-patched/byte-buddy-dep/src/main/java/net/bytebuddy/dynamic/ClassFileLocator.java	2023-01-12 09:04:56.000000000 -0800
@@ -476,7 +476,29 @@
              * {@inheritDoc}
              */
             public ClassLoader run() {
-                return new URLClassLoader(new URL[0], ClassLoadingStrategy.BOOTSTRAP_LOADER);
+                return new BootLoaderProxy(new URL[0], ClassLoadingStrategy.BOOTSTRAP_LOADER);
+            }
+
+            /**
+             * A URLClassLoader which exposes addURL to be used as a Bootloader Proxy.
+             */
+            class BootLoaderProxy extends URLClassLoader {
+                /**
+                 * Constructs a new BootLoaderProxy for the given URLs and parent ClassLoader.
+                 * @param      urls the URLs from which to load classes and resources
+                 * @param      parent the parent class loader for delegation
+                 */
+                BootLoaderProxy(URL[] urls, ClassLoader parent) {
+                    super(urls, parent);
+                }
+
+                /**
+                 * {@inheritDoc}
+                 */
+                @Override
+                public void addURL(URL url) {
+                    super.addURL(url);
+                }
             }
         }
 
